homeassistant/components/group/entity.py

```diff
@@ -425,10 +430,34 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
         elif tr_state.attributes.get(ATTR_ASSUMED_STATE):
             self._assumed_state = True
 
+        registry: GroupIntegrationRegistry = self.hass.data[REG_KEY]
+
+        def _update_active_on_off_states() -> tuple[set[str], set[str]]:
```

Comment:
Please add a docstring which explains what this does

Reply:
Refactored to inline code, I'll add some comment